modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/object/definitions/portlet/action/GetObjectFieldDeleteInfoMVCResourceCommand.java

```diff
@@ -125,22 +111,21 @@ private boolean _shouldShowDeletionModal(
 		if (objectDefinition.isApproved() &&
 			_shouldDeleteLastPublishedObjectDefinitionObjectField(
 				objectDefinition, objectField) &&
-			_shouldDeleteObjectFieldObjectValidationRuleSetting(
-				objectDefinition, objectField)) {
+			_shouldDeleteObjectFieldObjectValidationRuleSetting(objectField)) {
```

Comment:
Why this validation is called here and [here](https://github.com/liferay-objects/liferay-portal/pull/2732/commits/25286e0e33e5ac2590e7f6d4c7ccdef947b3a359#diff-c95ef5243d846bc93565258ce962f676027df08193eeeb24f8d9dcf2bada3e97R61-R62)

Reply:
Because of front-end needs. Needed to show a different modal. But I noticed what you are questioning.
Do you wanna now if I can call it only once and use the answer twice, right?